tsl/src/data_node.c

```diff
@@ -243,6 +243,17 @@ create_foreign_server(const char *const node_name, const char *const host, int32
 	return true;
 }
 
+TSConnection *
+data_node_get_connection_nothrow(const char *const data_node, char **errmsg)
+{
+	const ForeignServer *server;
+
+	Assert(data_node != NULL);
+	server = data_node_get_foreign_server(data_node, ACL_NO_CHECK, false, false);
+
+	return remote_connection_open_nothrow(server->serverid, GetUserId(), errmsg);
+}
```

Comment:
@65278 why aren't we using the existing `remote_connection_ping` API here?

Reply:
Hmm, self answer, because we want to send the healthcheck query to run on the DN itself.